### DISCUSSION TITLE
8

### DISCUSSION THREAD

**Discussion by Incog-1** [postalserver/postal#2899] [Development discussions]
Postal

**Comment by willpower232** [comment]
It sounds like your objects cover the events.

The `message_id` in the payload is the SMTP message ID.

The `id` in the payload refers to the numeric ID of the message within Postals database. You'll need that number to query the API for more information about the message.

Ideally, you would store the `id` which is returned when sending the message to Postal via the API and then you can be 100% sure which message the event is referring to. I'm not fully sure off the top of my head if the SMTP message ID is reliable as a unique message identifier in this context.

**Reply by willpower232** [reply]
It sounds like your software isn't handling the response very well so you'd have to see what else your code is doing

**Reply by willpower232** [reply]
`-1` suggests your code isn't exiting properly to send a successful response back to Postal, it should be like this

![image](https://github.com/postalserver/postal/assets/1619102/04730def-4f73-40b0-8f15-cee4bdf38bf1)

**Reply by willpower232** [reply]
They should all match MessageSent payload as they are just basic message status changes with a different status as necessary. The others are the ones that vary and have different payloads.

**Reply by willpower232** [reply]
I believe events are sent one at a time when they happen. I don't believe multiple can be triggered at the same time so no need for it to batch them up.

**Reply by willpower232** [reply]
Do you have click and open tracking set up for the domain you're sending from?

**Reply by willpower232** [reply]
Does Postal know your message has been opened? Can you see it on the Activity tab?

![image](https://github.com/postalserver/postal/assets/1619102/61e9b3df-85dd-44bd-b4a0-b24ce20418a0)

**Reply by willpower232** [reply]
if it isn't listed there then Postal doesn't know the message has been opened so wouldn't send the webhook, might be worth sending yourself a message with open tracking to see if you can spot the problem
